### ISSUE TITLE
[ANDROID] expo-splash-screen crashes when pressing reload in debug mode.

### ISSUE THREAD

**Issue by sarmad1995** [expo/expo#7821]
### Environment

Expo CLI 3.17.11 environment info:
    System:
      OS: macOS 10.15.3
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 13.3.0 - /usr/local/bin/node
      Yarn: 1.21.1 - /usr/local/bin/yarn
      npm: 6.13.2 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.5977832
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~37.0.3 => 37.0.3 
      react: ~16.9.0 => 16.9.0 
      react-native: ~0.61.4 => 0.61.5 
    npmGlobalPackages:
      expo-cli: 3.17.11

### Steps to Reproduce

Install expo splash screen on android 
Do all the configuration as in documentation
In development mode (debug) open the menu and click reload
### Expected Behavior

Should reload without any problem
### Actual Behavior

It crashes on reload press,
I also have some logs that might help.
```
java.lang.IllegalStateException: The specified child already has a parent. You must call removeView() on the child's parent first.
        at android.view.ViewGroup.addViewInner(ViewGroup.java:5106)
        at android.view.ViewGroup.addView(ViewGroup.java:4935)
        at android.view.ViewGroup.addView(ViewGroup.java:4875)
        at android.view.ViewGroup.addView(ViewGroup.java:4848)
        at expo.modules.splashscreen.SplashScreenController$showSplashScreen$2.run(SplashScreenController.kt:33)
        at android.app.Activity.runOnUiThread(Activity.java:6904)
        at expo.modules.splashscreen.SplashScreenController.showSplashScreen(SplashScreenController.kt:32)
        at expo.modules.splashscreen.SplashScreenController.showSplashScreen$default(SplashScreenController.kt:31)
        at expo.modules.splashscreen.SplashScreenController$handleRootView$1.onChildViewRemoved(SplashScreenController.kt:116)
        at android.view.ViewGroup.dispatchViewRemoved(ViewGroup.java:5007)
        at android.view.ViewGroup.removeAllViewsInLayout(ViewGroup.java:5650)
        at android.view.ViewGroup.removeAllViews(ViewGroup.java:5589)
        at com.facebook.react.ReactInstanceManager.clearReactRoot(ReactInstanceManager.java:723)
        at com.facebook.react.ReactInstanceManager.tearDownReactContext(ReactInstanceManager.java:1112)
        at com.facebook.react.ReactInstanceManager.runCreateReactContextOnNewThread(ReactInstanceManager.java:921)
        at com.facebook.react.ReactInstanceManager.recreateReactContextInBackground(ReactInstanceManager.java:908)
        at com.facebook.react.ReactInstanceManager.onReloadWithJSDebugger(ReactInstanceManager.java:878)
        at com.facebook.react.ReactInstanceManager.access$100(ReactInstanceManager.java:125)
        at com.facebook.react.ReactInstanceManager$2.onReloadWithJSDebugger(ReactInstanceManager.java:278)
        at com.facebook.react.devsupport.DevSupportManagerImpl.reloadJSInProxyMode(DevSupportManagerImpl.java:994)
        at com.facebook.react.devsupport.DevSupportManagerImpl.handleReloadJS(DevSupportManagerImpl.java:858)
        at com.facebook.react.devsupport.DevSupportManagerImpl$6.onOptionSelected(DevSupportManagerImpl.java:448)
        at com.facebook.react.devsupport.DevSupportManagerImpl$16.onClick(DevSupportManagerImpl.java:613)
        at com.android.internal.app.AlertController$AlertParams$3.onItemClick(AlertController.java:1222)
        at android.widget.AdapterView.performItemClick(AdapterView.java:330)
        at android.widget.AbsListView.performItemClick(AbsListView.java:1190)
        at android.widget.AbsListView$PerformClick.run(AbsListView.java:3198)
        at android.widget.AbsListView$3.run(AbsListView.java:4116)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```

### Update

Happened on release build as well, crashes on launch

**Comment by brentvatne** [comment]
hi! we need a reproducible example to help with this. please create a new issue and provide an example that we can run to demonstrate the issue. this is likely a bug in your own code and trying to provide a minimal reproducible example will help you to locate that bug! if you can reproduce it then it'll help us to investigate! more guidance on how to do this if you need it: https://stackoverflow.com/help/minimal-reproducible-example

https://github.com/brentvatne/splash-example - works as expected in this repo

**Comment by sarmad1995** [comment]
@brentvatne 
Any idea how can i resolve this issue? or where to look?
according to me I have done everything in the docs.

**Comment by brentvatne** [comment]
@sarmad1995 - you should do what i suggested above, create a minimal reproducible example. if you do this you will likely find the error. if you don't find the error that you made then you can share that minimal example and i can point it out to you, or i can see where there is a bug if there is one

**Comment by SamyOptimize** [comment]
@brentvatne I'm having the same problem as well, I've made a brand new `expo init` project and chose the `tabs template` and `ejected to bare work flow` and added the `expo-splash-screen` and set it up for android as the instructions suggested. When I reload the app in Debug mode I get the exact same error, and when running the app in release mode the error will keep showing up and the app keeps crashing

**Comment by brentvatne** [comment]
@SamyOptimize - can you share the project on github please?

**Comment by SamyOptimize** [comment]
@brentvatne  Here https://github.com/SamyOptimize/expo-splash-demo 
I haven't done the iOS part because I have it set up on my original project and it's working fine for iOS

**Comment by SamyOptimize** [comment]
Okay the problem with the release build on my original project was because i forgot to publish/export some of the native changes i made but the problem still persists when reloading during debug mode

**Comment by brentvatne** [comment]
@SamyOptimize 

1. you can't import SplashScreen from the expo package in App.js after ejecting, need to switch it to import from expo-splash-screen
2. the project you provided doesn't build for android

**Comment by SamyOptimize** [comment]
Sorry about the incomplete repo, I was trying just to get the error to show up for android and didn't edit anything in App.js. In my other original project everything is imported just fine in my App.js, when i run the expo-splash-screen command and provide the .png and everything it would get stuck and never finish so I manually installed it based on [the warning](https://github.com/expo/expo/tree/master/packages/expo-splash-screen#automatic-configuration-1) provided in the Docs. The only thing i noticed is that if I follow everything in the manual installation for android, I would get the error to show up even if I don't use anything in my App.js related to expo-splash-screen (Commented it out to test). But if I comment this line of code  in my MainActivity: 
```
 @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
  // SplashScreen.show(this, SplashScreenImageResizeMode.CONTAIN, ReactRootView.class);
  }
```
The error won't show up and of course the splash screen won't show either.

I will try cleaning my project from anything related to splash screen for android and try again with the expo-splash-screen command and see if it fixes it.

**Comment by SamyOptimize** [comment]
Also I forgot to add, this also happens when reloading the app in release mode when using `Updates.reloadAsync()`

**Comment by NiciusB** [comment]
I don't have anything useful to add, but we're seeing the same crash report on the google playstore as the one in the initial bug report.
I'm unable to reproduce it, either by launching the app or using `Updates.reloadAsync()`

On iOS we do also have crash reports related to expo-splash-screen like the one reported in #7689

**Comment by SamyOptimize** [comment]
@brentvatne Any news this issue?

**Comment by alorr10** [comment]
I'm seeing this too in production on android

**Comment by alorr10** [comment]
So I was seeing this in production too and I _think_ I fixed it - I was conditionally rendering an initial route in react navigation based on an authenticated firebase user, and i had a loading component that checked for the user and returned null if there was no user. If there was a user, I returned a stack navigator. 

I was storing this user in state, but I didnt set a default value. It looked like `const [user, setUser] = React.useState()` and changing it to `const [user, setUser] = React.useState(null)` solved the crash for me.

TLDR: check all your javascript code for what you render before you hide the splash screen and make sure there aren't any little mistakes like that.

**Comment by picklenick-dev** [comment]
I think I fixed it as well. I had it in a setTimeout() to add some delay. Moved hiding the splashscreen and I havent had the same problem.

**Comment by harikrishnagit** [comment]
Thanks @alorr10 for the tip. That helped me as well.

**Comment by hassan1319** [comment]
@brentvatne solution worked for me

**Comment by hassan1319** [comment]
> @SamyOptimize
> 
>     1. you can't import SplashScreen from the expo package in App.js after ejecting, need to switch it to import from expo-splash-screen. i replaced the contents of App.js with [this](https://gist.github.com/brentvatne/b2cf87a7270b3f0d9b1dbc6f9b30b7f1) to isolate the example
> 
>     2. ~the project you provided doesn't build for android~ looks like the debug keystore wasn't included, copied it over and it builds. i ran `yarn expo-splash-screen -p "android" -r "cover" "#0a0234" assets/splashscreen_image.png`and rebuilt and it worked as expected including on reload
> 
>     3. the project you provided had no launchscreen set on ios - I ran `yarn expo-splash-screen -p "ios" -r "cover" "#0a0234" assets/splashscreen_image.png` and that fixes this problem
> 
>     4. once i confirmed this worked as expected, i replaced `import { SplashScreen } from 'expo';` with `import * as SplashScreen from 'expo-splash-screen';` and updated function calls `SplashScreen.hide()` to  `SplashScreen.hideAsync()` and `SplashScreen.preventAutoHide()` to `SplashScreen.preventAutoHideAsync()` and reloading works as expected [(updated App.js with these changes)](https://gist.github.com/brentvatne/13e5d207ed0093a34f53ac3b6e5671ee).
> 
> 
> we need to improve this flow so it just works out of the box when you eject, this is something we are actively working on. in the future, `expo-splash-screen` will be used in managed workflow as well, which will make this a lot easier because no code will need to change.


This soluton worked for me. Thanks!
